nemo/collections/asr/parts/utils/decoder_timestamps_utils.py

```diff
@@ -0,0 +1,761 @@
+
+            QuartzNet15x5Base-En
+            stt_en_citrinet*
+            stt_en_conformer_ctc_*
+
+        To assign a proper decoding function for generating timestamp output,
+        the name of .nemo file should include the architecture name such as:
+        'quartznet', 'conformer', and 'citrinet'.
+
+        decoder_delay_in_sec is the amount of delay that is compensated during the word timestamp extraction.
+        word_ts_anchor_offset is the reference point for a word and used for matching the word with diarization labels.
+        Each ASR model has different optimal decoder delay and word timestamp anchor offset.
+        To obtain an optimized diarization result with ASR, decoder_delay_in_sec and word_ts_anchor_offset
+        need to be searched on a development set.
+        """
+        if 'quartznet' in ASR_model_name.lower():
+            self.run_ASR = self.run_ASR_QuartzNet_CTC
+            self.encdec_class = EncDecCTCModel
+            self.decoder_delay_in_sec = if_none_get_default(self.params['decoder_delay_in_sec'], 0.04)
+            self.word_ts_anchor_offset = if_none_get_default(self.params['word_ts_anchor_offset'], 0.12)
+            self.asr_batch_size = if_none_get_default(self.params['asr_batch_size'], 4)
+            self.model_stride_in_secs = 0.02
+
+        elif 'conformer' in ASR_model_name.lower():
+            self.run_ASR = self.run_ASR_BPE_CTC
+            self.encdec_class = EncDecCTCModelBPE
+            self.decoder_delay_in_sec = if_none_get_default(self.params['decoder_delay_in_sec'], 0.08)
+            self.word_ts_anchor_offset = if_none_get_default(self.params['word_ts_anchor_offset'], 0.12)
+            self.asr_batch_size = if_none_get_default(self.params['asr_batch_size'], 16)
+            self.model_stride_in_secs = 0.04
+
+            # Conformer requires buffered inference and the parameters for buffered processing.
+            self.chunk_len_in_sec = 5
+            self.total_buffer_in_secs = 25
+
+        elif 'citrinet' in ASR_model_name.lower():
+            self.run_ASR = self.run_ASR_CitriNet_CTC
+            self.encdec_class = EncDecCTCModelBPE
+            self.decoder_delay_in_sec = if_none_get_default(self.params['decoder_delay_in_sec'], 0.16)
+            self.word_ts_anchor_offset = if_none_get_default(self.params['word_ts_anchor_offset'], 0.2)
+            self.asr_batch_size = if_none_get_default(self.params['asr_batch_size'], 4)
+            self.model_stride_in_secs = 0.08
+
+        else:
+            raise ValueError(f"Cannot find the ASR model class for: {self.params['ASR_model_name']}")
+
+        if ASR_model_name.endswith('.nemo'):
+            asr_model = self.encdec_class.restore_from(restore_path=ASR_model_name)
+        else:
+            asr_model = self.encdec_class.from_pretrained(model_name=ASR_model_name, strict=False)
+
+        if self.ctc_decoder_params['pretrained_language_model'] and 'pyctcdecode' in sys.modules:
```

Comment:
Best is to check if any LM passed, then check for if they installed pyctcdecode, if not raise error else proceed